testsuite/smoke-test/src/operational_tooling.rs

```diff
@@ -148,6 +148,52 @@ fn test_create_validator_lcs_file() {
     create_validator_with_file_writer(write_key_to_file_lcs_format);
 }
 
+#[test]
+fn test_disable_address_validation() {
+    let num_nodes = 1;
+    let (_env, op_tool, backend, _) = launch_swarm_with_op_tool_and_backend(num_nodes, 0);
+
+    // Try to set the validator config with a bad address and verify failure
+    let bad_network_address = NetworkAddress::from_str("/dns4/127.0.0.1/tcp/1234").unwrap();
+    op_tool
+        .set_validator_config(
+            Some(bad_network_address.clone()),
+            None,
+            &backend,
+            false,
+            false,
+        )
+        .unwrap_err();
+
+    // Now disable address verification to set the validator config with a bad network address
+    let txn_ctx = op_tool
+        .set_validator_config(Some(bad_network_address), None, &backend, false, true)
+        .unwrap();
+    assert_eq!(VMStatusView::Executed, txn_ctx.execution_result.unwrap());
+
+    // Rotate the consensus key and verify that it isn't blocked by a bad network address
+    let _ = op_tool.rotate_consensus_key(&backend, false).unwrap();
+
+    // Rotate the validator network key and verify that it isn't blocked by a bad network address
+    let _ = op_tool
+        .rotate_validator_network_key(&backend, false)
+        .unwrap();
+
+    // Rotate the fullnode network key and verify that it isn't blocked by a bad network address
+    let _ = op_tool
+        .rotate_fullnode_network_key(&backend, false)
+        .unwrap();
+
+    // Rotate the operator key and verify that it isn't blocked by a bad network address
+    let _ = op_tool.rotate_operator_key(&backend, false).unwrap();
+
+    // Update the validator network address with a valid address
+    let new_network_address = NetworkAddress::from_str("/ip4/10.0.0.16/tcp/80").unwrap();
+    let _ = op_tool
+        .set_validator_config(Some(new_network_address), None, &backend, false, false)
+        .unwrap();
+}
+
```

Comment:
I don't really think we need a smoke test for this, when we can get everything we need from an integration test.

Reply:
Yeah, I think that's fair, although we do check that transactions are being executed by the blockchain allowing us to set bad networking addresses. But for the other operations, we could check the tool works using integration/unit style testing. I'll leave this as-is for now though as it's simpler to check all this functionality in once place. Once we start to think about our testing principles we can take a look at all the tests to figure out where things should live.